.github/workflows/trunk-cli.yml

```diff
@@ -0,0 +1,77 @@
+name: Trunk CLI workflow
```

Comment:
This workflow runs linting, unit test, and a functional test that makes sure pgx-builder will build a pgx extension - adding so I make sure I do not break pgx builder while working on C builder.